gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsRecord.java

```diff
@@ -0,0 +1,292 @@
+      this.ttl = record.ttl;
+      this.type = record.type;
+      this.rrdatas.addAll(record.rrdatas);
+    }
+
+    /**
+     * Adds a record to the record set. The records should be as defined in RFC 1035 (section 5) and
+     * RFC 1034 (section 3.6.1). Examples of records are available in Google DNS documentation.
+     *
+     * @see <a href="https://cloud.google.com/dns/what-is-cloud-dns#supported_record_types">Google
+     * DNS documentation </a>.
+     */
+    public Builder addRecord(String record) {
+      this.rrdatas.add(checkNotNull(record));
+      return this;
+    }
+
+    /**
+     * Removes a record from the set. An exact match is required.
+     */
+    public Builder removerRecord(String record) {
+      this.rrdatas.remove(checkNotNull(record));
+      return this;
+    }
+
+    /**
+     * Removes a record on the given index from the set.
+     */
+    public Builder removerRecord(int index) {
+      checkArgument(index >= 0 && index < this.rrdatas.size(), "The index is out of bounds. An " +
+              "integer between 0 and " + (this.rrdatas.size() - 1) + " is required. The provided " +
+              "value was " + index + ".");
+      this.rrdatas.remove(index);
+      return this;
+    }
+
+    /**
+     * Removes all the records.
+     */
+    public Builder clearRecords() {
+      this.rrdatas.clear();
+      return this;
+    }
+
+    /**
+     * Replaces the current records with the provided list of records.
+     */
+    public Builder records(List<String> records) {
+      this.rrdatas = Lists.newLinkedList(checkNotNull(records));
+      return this;
+    }
+
+    /**
+     * Sets name for this DNS record set. For example, www.example.com.
+     */
+    public Builder name(String name) {
+      this.name = checkNotNull(name);
+      return this;
+    }
+
+    /**
+     * Sets the number of seconds that this record can be cached by resolvers. This number must be
+     * non-negative.
+     *
+     * @param ttl A non-negative number of seconds
+     */
+    public Builder ttl(int ttl) {
+      checkArgument(ttl >= 0, "TTL cannot be negative. The supplied value was " + ttl + ".");
+      this.ttl = ttl;
+      return this;
+    }
+
+    /**
+     * The identifier of a supported record type, for example, A, AAAA, MX, TXT, and so on.
+     */
+    public Builder type(DnsRecordType type) {
+      this.type = checkNotNull(type);
+      return this;
+    }
+
+    /**
+     * Builds the DNS record.
+     */
+    public DnsRecord build() {
+      return new DnsRecord(this);
+    }
+  }
+
+  DnsRecord(Builder builder) {
+    this.name = builder.name;
+    this.rrdatas = ImmutableList.copyOf(builder.rrdatas);
+    this.ttl = builder.ttl;
+    this.type = builder.type;
+  }
+
+  /**
+   * Creates a builder pre-populated with the attribute values of this instance.
+   */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  /**
+   * Creates an empty builder.
+   */
+  public static Builder builder() {
```

Comment:
We often make user-assigned mandatory fields for objects parameters to the builder function.  For example, since "name" is mandatory, you could make this function signature `public static Builder builder(String name)`.  In the function you could say `return new Builder().name(name);`